src/Identity/Core/src/SecurityStampValidator.cs

```diff
@@ -23,16 +23,52 @@ public class SecurityStampValidator<TUser> : ISecurityStampValidator where TUser
     /// <param name="signInManager">The <see cref="SignInManager{TUser}"/>.</param>
     /// <param name="clock">The system clock.</param>
     /// <param name="logger">The logger.</param>
+    [Obsolete("ISystemClock is obsolete, use TimeProvider instead.")]
     public SecurityStampValidator(IOptions<SecurityStampValidatorOptions> options, SignInManager<TUser> signInManager, ISystemClock clock, ILoggerFactory logger)
     {
         ArgumentNullException.ThrowIfNull(options);
         ArgumentNullException.ThrowIfNull(signInManager);
         SignInManager = signInManager;
         Options = options.Value;
+        Time = new TimeClockProvider(clock);
         Clock = clock;
         Logger = logger.CreateLogger(GetType());
     }
 
+    /// <summary>
+    /// Creates a new instance of <see cref="SecurityStampValidator{TUser}"/>.
+    /// </summary>
+    /// <param name="options">Used to access the <see cref="IdentityOptions"/>.</param>
+    /// <param name="signInManager">The <see cref="SignInManager{TUser}"/>.</param>
+    /// <param name="clock">The system clock.</param>
+    /// <param name="time">The system clock.</param>
+    /// <param name="logger">The logger.</param>
+    [Obsolete("ISystemClock is obsolete, use TimeProvider instead.")]
+    public SecurityStampValidator(IOptions<SecurityStampValidatorOptions> options, SignInManager<TUser> signInManager, ISystemClock clock, ILoggerFactory logger, TimeProvider time)
```

Comment:
This overload has both? And `ISystemClock` isn't used?

Reply:
Alternative idea: register the type in DI with a func to explicitly call the constructor we want.

```cs
services.AddScoped<IWhatever>(sp => new Whatever(sp.GetRequiredService<TimeProvider>());
```

Although this would be tricky with generic types.

Reply:
I can try that for the validators.

The auth handlers are registered generically.
https://github.com/dotnet/aspnetcore/blob/e21ecb3dc51338f4d8288a844cb8491f550e4cd1/src/Security/Authentication/Core/src/AuthenticationBuilder.cs#L51

Reply:
Ug, the validators are also generic, that's not going to work.